crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -175,7 +176,9 @@ impl ServerBuilder {
     pub fn build(self) -> Result<Server, Error> {
         let (address, schema, router) = self.build_components();
 
-        let app = router.layer(axum::extract::Extension(schema));
+        let app = router
+            .layer(axum::extract::Extension(schema))
+            .layer(CorsLayer::permissive());
```

Comment:
This is here and not in the `init_router` function as it did not seem to work there at all.

Reply:
I'm not really an expert here, but I think we should probably lock this down a bit more than this, e.g. something like:

https://github.com/MystenLabs/sui/blob/d67d8c2b85a7fd83be2599b1a13993860e3b2e17/crates/sui-json-rpc/src/lib.rs#L103-L114

which is how we configure JSON-RPC.

I think it's fine for us to only allow cross-origin POST requests (from any origin) with the headers that we expect, as long as direct GET requests (e.g. for GraphiQL) are not affected by that configuration.

Reply:
Yes, I've seen this yesterday too and thought that probably we should go the same route. Thanks for confirming it!

Reply:
I basically ended up using the same logic from JSON RPC. 
@hayes-mysten - this will work except custom headers will not be accepted, just the two headers we have now (`x-sui-rpc-show-usage` and `x-sui-rpc-version`). I think this should be fine, so let me know otherwise.